### PR TITLE
Add the broken testing framework

### PR DESCRIPTION
**Pull request by michipili**

### CODE REVIEW

**Comment by yallop** [comment]
The 3.12 build is [failing](https://travis-ci.org/ocaml/opam-repository/jobs/74868784):

```
### stderr ###
# File "broken.ml", line 257, characters 21-47:
# Error: Unbound value Filename.get_temp_dir_name
```

[Filename.get_temp_dir_name](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Filename.html#VALget_temp_dir_name) was added in 4.00.0.  Could you please add the following version constraint?

```
available: [ ocaml-version >= "4.00.0" ]
```

File: packages/broken/broken.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+dev-repo: "https://github.com/michipili/broken.git"
+tags: [
+  "test"
+  "unit"
+  "application"
+]
+build: [
+  ["./configure" "--prefix" prefix]
+  ["bmake" "-I%{bsdowl:share}%" "all"] {os != "freebsd"}
+  ["make"  "-I%{bsdowl:share}%" "all"] {os  = "freebsd"}
```
Comment (avsm) [inline packages/broken/broken.0.3.0/opam]: is the BSD make variant used here also compatible with OpenBSD?

**Comment by michipili** [comment]
The `opam-lint-error` is a false positive, because I use **bmake** instead of **GNU Make** and the variable is made for users of **GNU Make**.

@avsm I am not sure about this, this would need to be tested. The make macros I use have a comprehensive test-suite but I have no OpenBSD box ready here to test https://github.com/michipili/bsdowl

@yallop Yes, I added this line, thanks!

**Comment by yallop** [comment]
Merging for now.  We can revisit the OpenBSD issue later if necessary.

**Comment by michipili** [comment]
Thanks!

Reading the *man page* it looks to me that the **make** program in the base system of CURRENT is too old and that **bmake** should be installed. I will check this for sure.

**Pull request merged**